src/TraceEvent/EventPipe/NetTraceFormat_v6.md

```diff
@@ -0,0 +1,239 @@
+
+#### Updated semantics for Trace fields
+
+The TraceBlock has a ProcessId field that was historically set to the ID of the process being traced. When doing a trace that may contain more than one process this field should be set to zero.
+
+The TraceBlock PointerSize field represents the machine pointer size for traces that may contain more than one process. For processes whose bitness is smaller than the machine bitness (such as when using an emulator), memory addresses should be zero extended to the machine pointer size.
+
+The ExpectedCPUSamplingRate field has been unused for a long time. Writers are recommended to set its value to zero and readers are recommended to ignore it.
+
+## Updated type metadata and additional payload field types
+
+### Motivation
+
+For file format size efficiency and writing efficiency there are a handful of new event payload paramater encodings we'd like to support. Each new encoding requires a corresponding representation in type metadata so that the file format readers understand how to decode it. We also want to support optional message templates and a human readable event description so that tools can better describe events to engineers viewing the traces.
+Last, we are taking the opportunity to simplify the metadata encoding format. Currently there is a variety of unused, redundant, and inefficient data in the V5 metadata encoding.
+
+### Changes
+
+#### Updated encoding for MetadataBlock
+
+In V5 the MetadataBlock reused the format of the EventBlock but many of the fields were unused. V6 uses an updated streamlined format:
+
+- uint16 Count
+- Concatenated sequence of Count MetadataBlobs, each of which is:
+  - varuint MetaDataId;       // The Meta-Data ID that is being defined.
+  - string ProviderName;      // varuint length-prefixed UTF8 string (no null terminator)
+  - varuint EventId;          // A small number that uniquely represents this Event within this provider.  
+  - string EventName;         // varuint length-prefixed length prefixed UTF8 string (no null terminator)
+  - FieldDescriptions;        // a list of the payload field names and types - described below
+  - OptionalMetadata;         // a list of optional metadata properties that can be attached to an event - described below
+
+FieldDescriptions format is:
+
+- varuint Count
+- Contenated sequence of Count field descriptions, each of which is:
+  - Type;                     // The type of a field - described below
+  - string FieldName;         // varuint length-prefixed UTF8 string (no null terminator)
+
+Type format is:
+
+- uint8 TypeCode;             // Type is a discriminated union and this is the discriminant. 
+- optional Type ElementType;  // Only present if TypeCode == Array (19).
+- optional FieldDescriptions ObjectFields; // Only present if TypeCode == Object (1).
```

Comment:
In the case of more complex types (e.g. objects), the format seems to support nested objects (objects within objects).  I'd expect this to be supported given that this is used in TraceLogging events.  Just want to make sure that you do expect this to be supported.

Reply:
Yep, I'm expecting it to be supported. FieldDescriptions contains 1 or more Types, and a Type could recursively be an object with its own nested FieldDescriptions.

Reply:
Great!  Thanks.